src/dunst.c

```diff
@@ -72,12 +72,12 @@ void check_timeouts(void)
                         continue;
                 }
 
+                /* shift iter before we possibly delete the current notification */
+                iter = iter->next ? iter->next : NULL;
```

Comment:
My point is that `iter` *cannot* be null at this point because
a) the while loop checks if `iter` is NULL and since this is the only line where we modify `iter` there is no way for it to be null without it being a bug and 
b) We already access `iter->data` at the top of the loop, if somehow `iter` is null dunst would have crashed long before reaching this point.

In my opinion silently handling an impossible condition is a bad practice since since it can obscure bugs that might be introduced.